src/test/java/org/project36/qualopt/web/rest/EmailTemplateResourceIntTest.java

```diff
@@ -0,0 +1,274 @@
+        emailTemplateRepository.saveAndFlush(emailTemplate);
+        int databaseSizeBeforeUpdate = emailTemplateRepository.findAll().size();
+
+        // Update the email
+        EmailTemplate updatedEmailTemplate = emailTemplateRepository.findOne(emailTemplate.getId());
+
+        restEmailMockMvc.perform(put("/api/emailTemplates")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(updatedEmailTemplate)))
+            .andExpect(status().isOk());
+
+        // Validate the Email in the database
+        List<EmailTemplate> emailTemplateList = emailTemplateRepository.findAll();
+        assertThat(emailTemplateList).hasSize(databaseSizeBeforeUpdate);
+        EmailTemplate testEmailTemplate = emailTemplateList.get(emailTemplateList.size() - 1);
+        assertEquals(testEmailTemplate, emailTemplate);
+    }
+
+    @Test
+    @Transactional
+    public void updateNonExistingEmailTemplate() throws Exception {
+        int databaseSizeBeforeUpdate = emailTemplateRepository.findAll().size();
+
+        // Create the Email Template
+
+        // If the entity doesn't have an ID, it will be created instead of just being updated
+        restEmailMockMvc.perform(put("/api/emailTemplates")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(emailTemplate)))
+            .andExpect(status().isCreated());
+
+        // Validate the Email in the database
+        List<EmailTemplate> emailTemplateList = emailTemplateRepository.findAll();
+        assertThat(emailTemplateList).hasSize(databaseSizeBeforeUpdate + 1);
+    }
+
+    @Test
+    @Transactional
+    public void deleteEmailTemplate() throws Exception {
+        // Initialize the database
+        emailTemplateRepository.saveAndFlush(emailTemplate);
+        int databaseSizeBeforeDelete = emailTemplateRepository.findAll().size();
+
+        // Get the email
+        restEmailMockMvc.perform(delete("/api/emailTemplates/{id}", emailTemplate.getId())
+            .accept(TestUtil.APPLICATION_JSON_UTF8))
+            .andExpect(status().isOk());
+
+        // Validate the database is empty
+        List<EmailTemplate> emailTemplateList = emailTemplateRepository.findAll();
+        assertThat(emailTemplateList).hasSize(databaseSizeBeforeDelete - 1);
+    }
+
+    @Test
+    @Transactional
+    public void equalsVerifier() throws Exception {
+        TestUtil.equalsVerifier(Email.class);
+        EmailTemplate emailTemplate1 = new EmailTemplate();
+        emailTemplate1.setId(1L);
+        EmailTemplate emailTemplate2 = new EmailTemplate();
+        emailTemplate2.setId(emailTemplate1.getId());
+        assertThat(emailTemplate1).isEqualTo(emailTemplate2);
+        emailTemplate2.setId(2L);
+        assertThat(emailTemplate1).isNotEqualTo(emailTemplate2);
+        emailTemplate1.setId(null);
+        assertThat(emailTemplate1).isNotEqualTo(emailTemplate2);
+    }
+}
```

Comment:
Very good comprehensive tests 💯

Reply:
thanks but these were implemented by the original author, I cp and pasted here and changed the type basically. I'll take the credit :P